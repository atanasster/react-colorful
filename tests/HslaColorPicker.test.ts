import * as path from "path";
import { loadConfigurations } from "@component-controls/config";
import { renderExample } from "@component-controls/test-renderers";
import { render, act } from "@testing-library/react";
import { renderErr } from "@component-controls/test-renderers";

import doc, { Overview } from "./../docs/src/HslaColorPicker.stories";

describe("HslaColorPicker", () => {
  const configPath = path.resolve(__dirname, "../docs");
  const config = loadConfigurations(configPath);

  test("Overview", () => {
    const example = Overview;

    let rendered;
    act(() => {
      rendered = renderExample({
        example,
        doc,
        config
      });
    });
    if (!rendered) {
      renderErr();
      return;
    }
    const { asFragment } = render(rendered);
    expect(asFragment()).toMatchSnapshot();
  });
});
