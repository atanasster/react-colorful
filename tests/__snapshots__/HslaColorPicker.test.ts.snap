// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HslaColorPicker Overview 1`] = `
<DocumentFragment>
  <div
    class="react-colorful"
  >
    <div
      class="react-colorful__saturation"
      style="background-color: rgb(0, 0, 0);"
    >
      <div
        aria-label="Color"
        aria-valuetext="Saturation 67%, Brightness 24%"
        class="react-colorful__interactive"
        role="slider"
        tabindex="0"
      >
        <div
          class="react-colorful__pointer react-colorful__saturation-pointer"
          style="top: 76%; left: 66.66666666666666%;"
        >
          <div
            class="react-colorful__pointer-fill"
            style="background-color: rgb(20, 20, 20);"
          />
        </div>
      </div>
    </div>
    <div
      class="react-colorful__hue"
    >
      <div
        aria-label="Hue"
        aria-valuetext="213"
        class="react-colorful__interactive"
        role="slider"
        tabindex="0"
      >
        <div
          class="react-colorful__pointer react-colorful__hue-pointer"
          style="top: 50%; left: 59.166666666666664%;"
        >
          <div
            class="react-colorful__pointer-fill"
            style="background-color: rgb(0, 0, 0);"
          />
        </div>
      </div>
    </div>
    <div
      class="react-colorful__alpha react-colorful__last-control"
    >
      <div
        class="react-colorful__alpha-gradient"
      />
      <div
        aria-label="Alpha"
        aria-valuetext="50%"
        class="react-colorful__interactive"
        role="slider"
        tabindex="0"
      >
        <div
          class="react-colorful__pointer react-colorful__alpha-pointer"
          style="top: 50%; left: 50%;"
        >
          <div
            class="react-colorful__pointer-fill"
            style="background-color: rgba(20, 20, 20, 0.5);"
          />
        </div>
      </div>
    </div>
  </div>
  <div
    style="padding-top: 10px; font-weight: bold;"
  >
    {
  "h": 213,
  "s": 50,
  "l": 16,
  "a": 0.5
}
  </div>
</DocumentFragment>
`;
